RoR Final Catpston Project

firepicaso opened the pull request:
### BudgetLover App

## Features of the app:
This budget app allows the user to
- Create a user account
- Create new transactions to keep track of expenses
- Create new categories to classify your expenses by categories
- Delete existing transactions
- See the calculated total amount of expenses per category


Kindly asking for a review

MahmoudBakr23 commented on app/controllers/group_purchases_controller.rb:
```diff
@@ -0,0 +1,2 @@
+class GroupPurchasesController < ApplicationController
+end
```
- [ ] Please delete any unused files or code for a cleaner and more optimized codebase. 👍

MahmoudBakr23 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+  def show; end
+
+  def edit; end
+
+  def update; end
+end
```
- [ ] Same here as well, since you're using devise gem, you don't need a users_controller unless you intend to customize its functionalities. 👍

MahmoudBakr23 commented on app/views/groups/index.html.erb:
```diff
@@ -0,0 +1,36 @@
+              </div>
+            </div>
+          <% end %>
+        </li>
+      <% end %>
+    </ul>
+  </section>
+  <section class="below-form">
+    <%= link_to "Add a Category", new_group_path, class: "btn", data: { turbo: false }  %>
+  </section>
+</main>
```
- [ ] As per project requirements, I highly encourage you to stick to the given design as much as possible. Please modify the colors and positions of the elements. Ex: the add category button is a green button at the bottom of the mobile view taking the full width of it. 😉 

![RorBudgetApp - Google Chrome 12_9_2023 1_19_59 PM](https://github.com/firepicaso/ror-budget-app/assets/47625407/76b04d0f-018a-483a-b018-26d7a7fffa20)

MahmoudBakr23 commented on app/views/purchases/index.html.erb:
```diff
@@ -0,0 +1,40 @@
+        </li>
+      <% end %>
+    </ul>
+    <div class="cagetory-total">
+      <span>TOTAL PAYMENT</span>
+      <span>$<%= total_amount(@group) %></span>
+    </div>
+  </section>
+  <section class="below-form">
+    <%= link_to "Add a Transaction", new_group_purchase_path %>
+  </section>
+</main>
```
- [ ] Please apply the style changes in the transactions index page as well. For the mobile view. 👍 
![RorBudgetApp - Google Chrome 12_9_2023 1_26_01 PM](https://github.com/firepicaso/ror-budget-app/assets/47625407/d5771c41-8b50-4aec-847a-d87a49067309)

MahmoudBakr23 commented on spec/features/groups_spec.rb:
```diff
@@ -0,0 +1,52 @@
+    expect(page).to have_content('$10.0')
+  end
+
+  it 'shows the Add Category button' do
+    expect(page).to have_link('Add a Category', href: new_group_path)
+  end
+
+  context 'When I click on a Group name' do
+    it "redirects me to that Group's transactions page" do
+      click_link('Office')
+      expect(page).to have_current_path(group_purchases_path(@group1))
+    end
+  end
+
+  context 'When I click on a Add Group button' do
+    it 'redirects me to form that adds new Group' do
+      click_link('Add a Category')
+      expect(page).to have_current_path(new_group_path)
+    end
+  end
```
- [ ] There are 4 failed tests located in the spec/features/group_spec and spec/features/purchase_spec, please see them through and fix them. 🙏 
![pwsh in ror-budget-app 12_9_2023 1_29_18 PM](https://github.com/firepicaso/ror-budget-app/assets/47625407/e16f3c34-9a38-412f-89bb-eaa2971211e7)

MahmoudBakr23 reviewed (changes requested):
Hello @firepicaso 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the controllers' restful methods
- Great job with authentication so far
- Amazing work with the models' associations and validations
- Good work with the database setup, migrations, and the thumbs up for the routes
- Descriptive PR and README
- Nice presentation
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

firepicaso commented:
Thanks a lot for the valuable feedback! @MahmoudBakr23 

I have removed all the unnecessary files from the repository
Updated the UI to match the reference
Fixed the test files so they are all passing now.

HAMMAS-SALEEM commented on app/assets/stylesheets/application.css:
```diff
@@ -5,6 +5,7 @@
   --pure-white: white;
   --pure-black: black;
+  --button-green: #43B73A;
 }
```
| Microverse Button | Your Button |
|-|-|
| ![image](https://github.com/firepicaso/ror-budget-app/assets/93039359/c2710a37-88fe-4e8b-aeeb-b83bf8f9ee18) | ![image](https://github.com/firepicaso/ror-budget-app/assets/93039359/da51e4ae-80a5-429b-9fea-38c10dc22860) |
| Original Color |
| ![image](https://github.com/firepicaso/ror-budget-app/assets/93039359/5be67e3c-a66c-4223-812d-b170da077c86) |

- [ ] Great work on making the button well but it would be if you use the right color to follow the wireframe :+1: as suggested by the previous reviewer :+1: .
- [ ] It would be nice If you added the white inside border of the button which you can see in the wireframe to make your design look professional :blush: .

HAMMAS-SALEEM commented on README.md:
```diff
@@ -0,0 +1,182 @@
+<h3 align="center"><b>BudgetLover</b></h3>
```
**[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you make your own logo of your name and put that in the readme file at the top to present yourself as a brand in the job market or tech industry to catch the recruiters' eyes 👍

HAMMAS-SALEEM reviewed (changes requested):
# Required Changes :recycle:
![Ukhv](https://user-images.githubusercontent.com/93039359/247195854-e274f284-2e3a-423b-8939-7e4efb638ee0.gif)

## Hey @firepicaso :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements
✔️ Great work on fixing the tests 🧪
✔️ Great work on making the mobile version 📱

Good job on implementing some of the requested changes, There are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

firepicaso commented:
Thank you so much @HAMMAS-SALEEM for your feedback.

I fixed the button styling and color.

VICTOR-CHUKWU commented on README.md:
```diff
@@ -0,0 +1,182 @@
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [BudgetLover App](https://budgetlover.onrender.com)
```
- [ ][Optional] The design changes you made have not been affected on your live page, kindly try building your app again on render

firepicaso replied:
thank you so much @VICTOR-CHUKWU.

I will try to deploy again.

VICTOR-CHUKWU reviewed (approved):
Hi @firepicaso ,
# STATUS: APPROVED! 🥇 💯 
![done](https://media.giphy.com/media/l1uga8ObPBnIBoXUQ/giphy.gif)

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
